### PR TITLE
Fix(Core/Color): ClassColor

### PR DESCRIPTION
**Pull request by PkllonG**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:
It was extracted from the client and calculated to arrive at it
```
RAID_CLASS_COLORS = {
	["HUNTER"] = { r = 0.67, g = 0.83, b = 0.45 },
	["WARLOCK"] = { r = 0.58, g = 0.51, b = 0.79 },
	["PRIEST"] = { r = 1.0, g = 1.0, b = 1.0 },
	["PALADIN"] = { r = 0.96, g = 0.55, b = 0.73 },
	["MAGE"] = { r = 0.41, g = 0.8, b = 0.94 },
	["ROGUE"] = { r = 1.0, g = 0.96, b = 0.41 },
	["DRUID"] = { r = 1.0, g = 0.49, b = 0.04 },
	["SHAMAN"] = { r = 0.0, g = 0.44, b = 0.87 },
	["WARRIOR"] = { r = 0.78, g = 0.61, b = 0.43 },
	["DEATHKNIGHT"] = { r = 0.77, g = 0.12 , b = 0.23 },
};
```
```
	local classColor = classFilename and RAID_CLASS_COLORS[classFilename] or NORMAL_FONT_COLOR;
	nameLabel:SetFormattedText("|cff%.2x%.2x%.2x%s|r", classColor.r * 255, classColor.g * 255, classColor.b * 255, UnitName("party"..i));
```
```
        struct cs
        {
            double r;
            double g;
            double b;
        };

        static cs _cs[10] =
        {
            { 0.67, 0.83, 0.45 },
            { 0.58, 0.51,  0.79 },
            { 1.0, 1.0,  1.0 },
            { 0.96,  0.55,  0.73 },
            { 0.41,  0.8,  0.94 },
            { 1.0,  0.96,  0.41 },
            { 1.0,  0.49,  0.04 },
            { 0.0,  0.44,  0.87 },
            { 0.78,  0.61,  0.43 },
            { 0.77,  0.12,  0.23 }
        };

        for (uint8 i = 0; i < 10; ++i)
        {
            printf("%.2x%.2x%.2x \n", unsigned int(_cs[i].r * 255), unsigned int(_cs[i].g * 255), unsigned int(_cs[i].b * 255));
        }
```
```
aad372
9382c9
ffffff
f48cba
68ccef
fff468
ff7c0a
0070dd
c69b6d
c41e3a
```

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/shared/SharedDefines.h [hunk 1]
```diff
@@ -365,11 +365,11 @@ const uint32 ItemQualityColors[MAX_ITEM_QUALITY] =
...
 {
-    0xff40c040,
-    0xff808080,
+    0xff3fbf3f,
+    0xff7f7f7f,
     0xffffff00,
-    0xffff8040,
-    0xffff2020
+    0xffff7f3f,
+    0xffff1919
```
Comment (PkllonG) [inline src/server/shared/SharedDefines.h:372]:
```
    struct HyperlinkColor
    {
        HyperlinkColor(uint32 c) : r(c >> 16), g(c >> 8), b(c), a(c >> 24) {}
        uint8 const r, g, b, a;
        bool operator==(uint32 c) const
        {
            if ((c & 0xff) ^ b)
                return false;
            if (((c >>= 8) & 0xff) ^ g)
                return false;
            if (((c >>= 8) & 0xff) ^ r)
                return false;
            if ((c >>= 8) ^ a)
                return false;
            return true;
        }
    };
```
About "quest" needs to be tested here

**Comment by sudlud** [comment]
Can you please provide information on how to test these changes?

Thanks!

**Comment by PkllonG** [comment]
I've tested it, and there's no code referencing this

**Comment by sudlud** [comment]
`.gear stats` seems to use the function

**Comment by sudlud** [comment]
what about another PR to use these colors within Chat.h -> playerLink()?

**Comment by sudlud** [comment]
I'll close this PR for now, for the reason in my comment above.

IMO when executed correctly this PR would not propose any changes at all as the colors are identical to the client values.

**Pull request closed without merging**